### DISCUSSION TITLE
Error code 23

### DISCUSSION THREAD

**Discussion by rkosick** [outdoorbits/little-backup-box#258] [General]
While doing rsync backup, I get an error in some situations when there are already files and it is a partial backup. Not sure what the exact circumstances are, that trigger this error.

The error I get:
"Exception: Partial transfer du to error, Error code '23'"

After 5 retries, none of the new files is copied. 
Lost source device is always false, source is always mounted. What is this error code?

**Comment by rkosick** [comment]
Oh I found the error codes and error code 23 is "Partial transfer due to error".

I get something like this:

2023-12-02 09:30:11
mount USB device: --uuid A280-44C9

2023-12-02 09:30:11
MOUNTED?: '/media/local_source' = True

2023-12-02 09:30:11
Backup from usb: 

2023-12-02 09:30:11
MOUNTED?: '/media/local_source' = True

2023-12-02 09:30:11
Lost source device usb? False

2023-12-02 09:30:11
Working...

2023-12-02 09:30:11
sshpass -p *** rsync -avh --info=FLIST0,PROGRESS2 --stats --no-owner --no-group --no-perms --mkpath --min-size=1 --exclude *.id --exclude images.db --compress --exclude *tims/ --dry-run /media/local_source/ [rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231106174243-16322](mailto:rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231106174243-16322)

2023-12-02 09:30:11
grep Number of regular files transferred

2023-12-02 09:30:13
Files to sync from folder '': 116

2023-12-02 09:30:13
Files to sync before backup: 116

2023-12-02 09:30:14
USB storages 
 > rsync server 
0 of 116
Time: ?
comparing
VPN: WireGuard

2023-12-02 09:30:14
sshpass -p *** rsync -avh --info=FLIST0,PROGRESS2 --stats --no-owner --no-group --no-perms --mkpath --min-size=1 --exclude *.id --exclude images.db --compress --exclude *tims/ /media/local_source/ [rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231106174243-16322](mailto:rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231106174243-16322)

2023-12-02 09:30:15
sshpass -p *** rsync -avh --info=FLIST0,PROGRESS2 --stats --no-owner --no-group --no-perms --mkpath --min-size=1 --exclude *.id --exclude images.db --compress --exclude *tims/ --dry-run /media/local_source/ [rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231106174243-16322](mailto:rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231106174243-16322)

2023-12-02 09:30:15
grep Number of regular files transferred

2023-12-02 09:30:15
Files to sync from folder '': 116

2023-12-02 09:30:15
Files missing: 116 files not synced.

2023-12-02 09:30:15
Files missing:
> lsblk -p -P -o PATH,MOUNTPOINT,UUID,FSTYPE
b'PATH="/dev/sda" MOUNTPOINT="" UUID="" FSTYPE=""\nPATH="/dev/sda1" MOUNTPOINT="/media/local_source" UUID="A280-44C9" FSTYPE="exfat"\nPATH="/dev/mmcblk0" MOUNTPOINT="" UUID="" FSTYPE=""\nPATH="/dev/mmcblk0p1" MOUNTPOINT="/boot" UUID="0B22-2966" FSTYPE="vfat"\nPATH="/dev/mmcblk0p2" MOUNTPOINT="/" UUID="3ad7386b-e1ae-4032-ae33-0c40f5ecc4ac" FSTYPE="ext4"\n'

2023-12-02 09:30:16
No abnormal conditions have been detected.

2023-12-02 09:30:16
Exception: Partial transfer due to error, Error code '23'

2023-12-02 09:30:16
No abnormal conditions have been detected.

2023-12-02 09:30:16
MOUNTED?: '/media/local_source' = True

2023-12-02 09:30:16
Lost source device usb? False

2023-12-02 09:30:17
Try backup 2 of 5

**Comment by outdoorbits** [comment]
Error 23 is issued by rsync and is unfortunately the message for errors with many different causes.

It can occur due to incorrect file system rights or due to file system corruption. In the latter case, the system could possibly even be repaired using LLB's Web UI tools.

Insufficient storage space on the target drive can also lead to this error message.

Another known problem is special characters and umlauts in file or folder names. If the source and target file systems are not identical or do not use the same character set, rsync may repeatedly see deviations during the comparison and therefore fail. This problem seems to occur regularly between Mac OS and Linux file systems. The solution here is to use the same file system on the target drive as on the source drive. Otherwise, it's definitely not a bad idea to avoid problematic characters in file and folder names.

**Comment by rkosick** [comment]
Thanks for the info. I'll investigate it.

**Reply by outdoorbits** [reply]
I added the info above to the wiki. As soon as you know, would you please tell if this was correct and helpful?

**Reply by rkosick** [reply]
Hello,
I would love to give some feedback, but unfortunatelly my investigations didn't lead anywhere. The problem was not really reproducible. Somethimes it worked, sometimes not. I thought it is because of my special setup using a RP Zero 2W. Kind of a hardware bottleneck bacause CPU usage is pretty high. Therefore I setup a RP 4 with your case an the buttons and all the stuff. After intensive testing it works great with the exact same SW and storage devices. I didn't change anything. Then I tested it again with the RP Zero 2W and it works all the time now. Maybe I changed something regarding the servers rsync config but I also had these problems performing a cloud service backup to dropbox. So not sure what the problem was.
Since I printed your case and did everything with the buttons I have some suggestings regarding documentation to improve:
First I don't know what the "lbb-case-mounting material washer x8.stl"  are for. Works without them and since they are tiny I'm ok with it.
I'm pretty sure that for the standard pins for the buttons (29, 31, 11, 13) up and down and left and right is mixed up. Up should be down and left schould be right and vice versa. But this was not hard to figure out.
Great job for designing the case!

The main problem right now with the buttons is that rsync backup is not behaving the way it does when it is started via web UI.
Starting it via web UI everything is fine:

----------------------------------------------------
2023-12-12 18:41:30
pre mount usb, source (device list changed)
PATH="/dev/sda" MOUNTPOINT="" UUID="" FSTYPE=""
PATH="/dev/sda1" MOUNTPOINT="" UUID="6FFA-70CD" FSTYPE="exfat"


2023-12-12 18:41:30
DeviceChosenIdentifier='--uuid 6FFA-70CD' ('/dev/sda1') prepared to mount at '/media/local_source', exfat

2023-12-12 18:41:30
MOUNTED?: '/media/local_source' = False

2023-12-12 18:41:31
mounting:
USB

2023-12-12 18:41:31
mount --uuid 6FFA-70CD /media/local_source -o uid=33,gid=33,umask=0

2023-12-12 18:41:31
mounted USB '--uuid 6FFA-70CD' at '/media/local_source': Msg.='$'

2023-12-12 18:41:31
MOUNTED?: '/media/local_source' = True

2023-12-12 18:41:31
Source o.k.
size: 244G
used: 115M
free: 244G
FS: exfat
 0.0% ____________________

2023-12-12 18:41:31
mount USB device: --uuid 6FFA-70CD

2023-12-12 18:41:31
MOUNTED?: '/media/local_source' = True

2023-12-12 18:41:31
Backup from usb: 

2023-12-12 18:41:31
MOUNTED?: '/media/local_source' = True

2023-12-12 18:41:31
Lost source device usb? False

2023-12-12 18:41:31
Working...

2023-12-12 18:41:31
sshpass -p *** rsync -avh --info=FLIST0,PROGRESS2 --stats --no-owner --no-group --no-perms --mkpath --min-size=1 --exclude *.id --exclude images.db --compress --exclude *tims/ --dry-run /media/local_source/ rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231211172327-48454

2023-12-12 18:41:31
grep Number of regular files transferred

2023-12-12 18:41:33
Files to sync from folder '': 4

2023-12-12 18:41:33
Files to sync before backup: 4

2023-12-12 18:41:33
USB storages 
 > rsync server 
0 of 4
Time: ?
comparing
VPN: WireGuard

2023-12-12 18:41:33
sshpass -p *** rsync -avh --info=FLIST0,PROGRESS2 --stats --no-owner --no-group --no-perms --mkpath --min-size=1 --exclude *.id --exclude images.db --compress --exclude *tims/ /media/local_source/ rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231211172327-48454

2023-12-12 18:41:52
USB storages 
 > rsync server 
2 of 4, 447.18kB/s
Time: 0:00:19
 50.0% >>>>>>>>>>__________
VPN: WireGuard
AND SO ON...
----------------------------------------------------

Starting it with the butons I get this:

----------------------------------------------------
2023-12-12 18:35:40
pre mount usb, target (device list changed)
PATH="/dev/sda" MOUNTPOINT="" UUID="" FSTYPE=""
PATH="/dev/sda1" MOUNTPOINT="" UUID="6FFA-70CD" FSTYPE="exfat"


2023-12-12 18:35:40
DeviceChosenIdentifier='--uuid 6FFA-70CD' ('/dev/sda1') prepared to mount at '/media/local_target', exfat

2023-12-12 18:35:40
MOUNTED?: '/media/local_target' = False

2023-12-12 18:35:40
pre mount usb, source (device list changed)
PATH="/dev/sda" MOUNTPOINT="" UUID="" FSTYPE=""
PATH="/dev/sda1" MOUNTPOINT="" UUID="6FFA-70CD" FSTYPE="exfat"


2023-12-12 18:35:40
DeviceChosenIdentifier='--uuid 6FFA-70CD' ('/dev/sda1') prepared to mount at '/media/local_source', exfat

2023-12-12 18:35:40
MOUNTED?: '/media/local_source' = False

2023-12-12 18:35:40
mounting:
USB

2023-12-12 18:35:40
mount --uuid 6FFA-70CD /media/local_target -o uid=33,gid=33,umask=0

2023-12-12 18:35:41
mounted USB '--uuid 6FFA-70CD' at '/media/local_target': Msg.='$'

2023-12-12 18:35:41
MOUNTED?: '/media/local_target' = True

2023-12-12 18:35:41
mounting:
USB

2023-12-12 18:35:41
mount --uuid 6FFA-70CD /media/local_source -o uid=33,gid=33,umask=0

2023-12-12 18:35:41
mounted USB '--uuid 6FFA-70CD' at '/media/local_source': Msg.='$'

2023-12-12 18:35:41
MOUNTED?: '/media/local_source' = True

2023-12-12 18:35:41
Target o.k.
size: 244G
used: 115M
free: 244G
FS: exfat
 0.0% ____________________

2023-12-12 18:35:41
mount USB device: --uuid 6FFA-70CD

2023-12-12 18:35:41
MOUNTED?: '/media/local_target' = False

2023-12-12 18:35:41
Source o.k.
size: 244G
used: 115M
free: 244G
FS: exfat
 0.0% ____________________

2023-12-12 18:35:41
mount USB device: --uuid 6FFA-70CD

2023-12-12 18:35:41
MOUNTED?: '/media/local_source' = True

2023-12-12 18:35:41
mount device: usb, source; WaitForDevice=True; DeviceIdentifierPresetThis=; DeviceIdentifierPresetOther=

2023-12-12 18:35:41
MOUNTED?: '/media/local_source' = True

2023-12-12 18:35:41
Backup from usb: 

2023-12-12 18:35:41
MOUNTED?: '/mnt/tech_source' = False

2023-12-12 18:35:41
MOUNTED?: '/media/local_source' = True

2023-12-12 18:35:41
Lost source device usb? False

2023-12-12 18:35:42
Working...

2023-12-12 18:35:42
sshpass -p *** rsync -avh --info=FLIST0,PROGRESS2 --stats --no-owner --no-group --no-perms --mkpath --min-size=1 --exclude *.id --exclude images.db --compress --exclude *tims/ --dry-run /media/local_source/ rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231211172327-48454

2023-12-12 18:35:42
grep Number of regular files transferred

2023-12-12 18:35:42
unmounting
USB-source

2023-12-12 18:35:43
Files to sync from folder '': 5

2023-12-12 18:35:43
Files to sync before backup: 5

2023-12-12 18:35:43
USB storages 
 > rsync server 
0 of 5
Time: ?
comparing
VPN: WireGuard

2023-12-12 18:35:43
sshpass -p *** rsync -avh --info=FLIST0,PROGRESS2 --stats --no-owner --no-group --no-perms --mkpath --min-size=1 --exclude *.id --exclude images.db --compress --exclude *tims/ /media/local_source/ rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231211172327-48454

2023-12-12 18:35:44
sshpass -p *** rsync -avh --info=FLIST0,PROGRESS2 --stats --no-owner --no-group --no-perms --mkpath --min-size=1 --exclude *.id --exclude images.db --compress --exclude *tims/ --dry-run /media/local_source/ rsync://lbb@192.168.151.151:873/little-backup-box/lbb_20231211172327-48454

2023-12-12 18:35:44
grep Number of regular files transferred

2023-12-12 18:35:45
Files to sync from folder '': 0

2023-12-12 18:35:45
MOUNTED?: '/media/local_source' = False

2023-12-12 18:35:45
Lost source device usb? True

2023-12-12 18:35:45
Lost device:
>lsblk -p -P -o PATH,MOUNTPOINT,UUID,FSTYPE
PATH="/dev/sda" MOUNTPOINT="" UUID="" FSTYPE=""
PATH="/dev/sda1" MOUNTPOINT="/media/local_target" UUID="6FFA-70CD" FSTYPE="exfat"
PATH="/dev/mmcblk0" MOUNTPOINT="" UUID="" FSTYPE=""
PATH="/dev/mmcblk0p1" MOUNTPOINT="/boot" UUID="37F9-0715" FSTYPE="vfat"
PATH="/dev/mmcblk0p2" MOUNTPOINT="/" UUID="cec43d98-fade-4285-8755-e55ea76456c5" FSTYPE="ext4"


2023-12-12 18:35:45
No abnormal conditions have been detected.

2023-12-12 18:35:46
Try backup 2 of 5

2023-12-12 18:35:46
umount: /media/local_source? 

2023-12-12 18:35:46
finished

2023-12-12 18:35:46
Ready
Insert sources

2023-12-12 18:35:46
pre mount usb, source (device list changed)
PATH="/dev/sda" MOUNTPOINT="" UUID="" FSTYPE=""


2023-12-12 18:35:48
online, IP:
192.168.245.76
192.168.151.202

2023-12-12 18:36:03
Lost device: pre remount:
>lsblk -p -P -o PATH,MOUNTPOINT,UUID,FSTYPE
PATH="/dev/sda" MOUNTPOINT="" UUID="" FSTYPE=""
PATH="/dev/sda1" MOUNTPOINT="/media/local_target" UUID="6FFA-70CD" FSTYPE="exfat"
PATH="/dev/mmcblk0" MOUNTPOINT="" UUID="" FSTYPE=""
PATH="/dev/mmcblk0p1" MOUNTPOINT="/boot" UUID="37F9-0715" FSTYPE="vfat"
PATH="/dev/mmcblk0p2" MOUNTPOINT="/" UUID="cec43d98-fade-4285-8755-e55ea76456c5" FSTYPE="ext4"
----------------------------------------------------

That behaviour was reproducible many times. It always works when using the web UI. But maybe again a coincidence?

**Reply by rkosick** [reply]
After waiting some time before connecting the source since I had something to do, starting it with the buttons worked, And I was not connected via web UI at the same time. Maybe there is something?

**Reply by outdoorbits** [reply]
Hard to say...??? 
By the way, I recognised to prefer the same direction to use as you do. That's why there is an option to turn display and buttons. 😀

**Reply by rkosick** [reply]
:)

**Reply by rkosick** [reply]
I'm at the end of another vacation. Everything worked perfect. Except starting the backup process with the buttons. Rsync backup and cloud backup do not work. But both work if initiating over the web browser. I can provide some logfiles when I'm back home on Sunday. Starting a local backup with the buttons works on the other side.

**Comment by outdoorbits** [comment]
I just tried a button initiated backup from usb to rsync without problems. Could you please send your log?

**Reply by rkosick** [reply]
So I had the option for automatic backup active. This seems to be a problem when trying to perform rsync or dropbox backup with the buttons.
With this option active I get the following log for rclone backup via dropbox (and rsync):

-----------------------------------------------------------------------------------------------

[5dropbox_buttons_fehler.log](https://github.com/outdoorbits/little-backup-box/files/13853145/5dropbox_buttons_fehler.log)

-----------------------------------------------------------------------------------------------

Without this option, rsync backup to my server works fine with the buttons. Dropbox works with the first backup.
But partial Backup always backup all the files and for every try, so like a loop:

-----------------------------------------------------------------------------------------------

[7dropbox_buttons_partiell_without_default_backup_loop.log](https://github.com/outdoorbits/little-backup-box/files/13853148/7dropbox_buttons_partiell_without_default_backup_loop.log)

-----------------------------------------------------------------------------------------------

For dropbox this also happens if I start the backup via web interface. But this seems to be a misconfiguration in rclone with dropbox I guess.

So there was probably only an error when the option for automatic backup is active. And only if you use the buttons?

**Reply by rkosick** [reply]
I don't see any misconfiguration in rclone with dropbox. Strange. Going to try google drive.

**Comment by outdoorbits** [comment]
For both there is an idea...
1.: The loop probem:
Are there files having special characters in the filename or other unusual names like starting with space or something like that?

2. In the other case it seems to run both backups at once (automatic: usb-usb and the button driven). This might be an error. I need to investigate.

**Reply by rkosick** [reply]
For 1: it should be the same problem performing local ISB-USB backup? But it isn't.

**Comment by outdoorbits** [comment]
Problem 2 should be solved, would you check?

**Comment by outdoorbits** [comment]
Problem 1: Depends on the charset of the filesystems. It your local fs have compatible charsets, it might work. Are there special characters?

**Reply by rkosick** [reply]
This seems to be a problem with Dropbox, at least for me now. Cloud backup to google drive works great.

**Comment by rkosick** [comment]
Did the update.
Works now with active automatic backup: usb-usb and the buttons! Perfect.
But only via Web Interface. 

Now after starting I get the QR Code with IP adress and after use the buttons to start the backup I see the QR Code again and nothing else works.

**Comment by rkosick** [comment]
Btw with automatic backup and Web interace only shows
IMAGE=/var/www/little-backup-box/tmp/ip-qr-link-192-168-151-125_64_online.png
as last entry

**Comment by outdoorbits** [comment]
That's just the new IP info. What's about the log monitor?

**Comment by rkosick** [comment]
[QR_code.log](https://github.com/outdoorbits/little-backup-box/files/13854883/QR_code.log)

Over Web interface backup is initiated. With buttons nothing happens. Just seeing the QR Code again. Which is btw a cool option.
Dosn't matter if automatic backup oiption is active.

**Comment by outdoorbits** [comment]
It is fixed. The second error today was because the pre running backup 
wasn't  stopped. Now in display menu both processes, the stop command 
and the start command started shortly after each other - but as 
background processes. So the stopp command killed the following start 
command - got a few more grey hairs now. ;-)

**Comment by rkosick** [comment]
Haha, I have enough grey hairs from stuff like that. :) Going to test it tomorrow.

**Reply by rkosick** [reply]
Works now again. Thanks.
